cmd/provisioner/revivelvs.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/google/lvmd/commands"
+	"github.com/urfave/cli/v2"
+	"k8s.io/klog"
+)
+
+var (
+	envDirectory = "CSI_LVM_MOUNTPOINT"
+)
+
+func reviveLVsCmd() *cli.Command {
+	return &cli.Command{
+		Name: "revivelvs",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  flagVGName,
+				Usage: "Required. the name of the volumegroup",
+				Value: "csi-lvm",
+			},
+			&cli.StringFlag{
+				Name:    flagDirectory,
+				Usage:   "Required. the name of the directory to mount the lv",
+				EnvVars: []string{envDirectory},
+				Value:   "/tmp/csi-lvm",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			if err := reviveLVs(c); err != nil {
+				klog.Fatalf("Error reviving logical volumes: %v", err)
+				return err
+			}
+			// stay alive
+			for {
+				time.Sleep(time.Hour)
+			}
+		},
+	}
+}
+
+// reviveLVs scans for existing volumes which are not mounted correctly
+func reviveLVs(c *cli.Context) error {
+	klog.Infof("starting reviver: %s", c)
+	vgName := c.String(flagVGName)
+	if vgName == "" {
+		return fmt.Errorf("invalid empty flag %v", flagVGName)
+	}
+	dirName := c.String(flagDirectory)
+	if dirName == "" {
+		return fmt.Errorf("invalid empty flag %v", flagDirectory)
+	}
+	vgexists := vgExists(vgName)
+	if !vgexists {
+		klog.Infof("volumegroup: %s not found\n", vgName)
```

Comment:
duplicate of the same code in createlv.go:236ff, we can create a func vgactivate ?

Reply:
good idea, done.